### PR TITLE
Fixes #168

### LINKED ISSUE #168: Non-replacing Scenes can't be popped
Something like:

```
def on_event(self, event, foo):
    event.scene.next = EphemeralScene
```

causes `EphemeralScene` to be started every time the current scene is returned to.

### PR DESCRIPTION
**Pull request by pathunstrom**

### CODE REVIEW

File: ppb/systems.py [hunk 1]
```diff
@@ -118,8 +118,30 @@ class Quitter(System):
...
+        self.count = count
+
     def activate(self, engine):
-        yield events.Quit()
+        self.counter += 1
+        if self.counter >= self.count:
+            yield events.Quit()
+
+
+class FailerSystem(System):
```
Comment (astronouth7303) [inline ppb/systems.py]: Why is this here and also in tests?

Comment (pathunstrom) [reply ppb/systems.py]: Mistake, half a refactor.
